onnxruntime/core/providers/nnapi/nnapi_builtin/builders/impl/resize_op_builder.cc

```diff
@@ -216,20 +216,22 @@ bool ResizeOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers
     }
 
     // scales
-    if (inputs.size() == 3 && !Contains(initializers, inputs[2].node_arg.Name())) {
+    bool using_scales = (inputs.size() == 3 && inputs[2].node_arg.Exists());
```

Comment:
```suggestion
    bool using_scales = (inputs.size() > 2 && inputs[2].node_arg.Exists());
```
is it possible for `inputs.size() == 4` and `inputs[3].node_arg.Exists() == false`?
I suppose it's also more flexible if another optional input is added in the future.

Reply:
There's no rule (afaik) in the ONNX spec saying missing optional outputs at the end of the inputs must not be provided, so the any test for an optional input must include a check on the name being empty (which is what Exists() does) if the input is present in the node definition.